[Form] [Form Theme] customize choice entry

gilles-g opened the pull request:
Hi,

I'm not sure if I have add my text on the right file..

My english is not good as you ;)

sudo-barun committed:
Fix example code of customization of bootstrapping in test

The change is copied from Symfony 4.4.

clemherreman committed:
[Messenger] Add doc for default routing for messages

gilles-g commented:
@javiereguiluz Any update about this?

KosticDusan4D committed:
Correcting bugs for provider and JSON login

On lines 501 and 502 I propose this change, because it confused me so much which provider to use, and in the case that you are explaining there you should use earlier configured app_user_provider, not users_in_memory.

On line 1043 I had a problem that $user is always null until I deleted the "?User" from parameters. I think that PHP gets confused here with the way it is written. And to say that now with that part removed everything works fine.

Thanks in advance and I hope that I contributed even a little as this is.

likeuntomurphy committed:
Change 'ignore' element to attribute in Serializer XML example

javiereguiluz committed:
[HttpKernel] Document the `terminate_on_cache_hit` option

ThomasLandauer committed:
Expanding recommendation for `finishView()`...

... to include the widgets created by ChoiceType as well.

Might conflict with a minor change I just did in https://github.com/symfony/symfony-docs/pull/17066

ThomasLandauer committed:
[Security] Adding info where login attempts are stored

BOB41K1987 committed:
[DependencyInjection] Add information about debugging tagged services

mpdude committed:
Document the `framework.router.cache_dir` setting

qzminski committed:
Fix the code example leading to an error

johnkrovitch committed:
[Security] Add documentation for programmatic login

johnkrovitch committed:
docs: add docs for programmatic logout

HeahDude committed:
[DependencyInjection] Document `stack` definitions

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  Trying to add quotation markup

xabbuh committed:
remove display names from envelope headers

OskarStark committed:
Merge branch '4.4' into 5.4

* 4.4:
  Fix use statement

OskarStark committed:
Merge branch '5.4' into 6.0

* 5.4:
  Fix use statement

OskarStark committed:
Merge branch '6.0' into 6.1

* 6.0:
  Fix use statement

OskarStark committed:
Merge branch '6.1' into 6.2

* 6.1:
  Fix use statement

OskarStark committed:
Merge branch '4.4' into 5.4

* 4.4:
  Remove reference to a removed configuration key

OskarStark committed:
Merge branch '5.4' into 6.0

* 5.4:
  Remove reference to a removed configuration key

OskarStark committed:
Merge branch '6.0' into 6.1

* 6.0:
  Remove reference to a removed configuration key

OskarStark committed:
Merge branch '6.1' into 6.2

* 6.1:
  Remove reference to a removed configuration key

OskarStark committed:
Merge branch '4.4' into 5.4

* 4.4:
  [Workflow] fix image labels

OskarStark committed:
Merge branch '5.4' into 6.0

* 5.4:
  [Workflow] fix image labels

OskarStark committed:
Merge branch '6.0' into 6.1

* 6.0:
  [Workflow] fix image labels

OskarStark committed:
Merge branch '6.1' into 6.2

* 6.1:
  [Workflow] fix image labels

javiereguiluz committed:
minor #17302 [Mailer] remove display names from envelope headers (xabbuh)

This PR was merged into the 5.4 branch.

Discussion
----------

[Mailer] remove display names from envelope headers

Commits
-------

69d06ffa6 remove display names from envelope headers

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  remove display names from envelope headers
  Fix use statement
  Remove reference to a removed configuration key
  [Workflow] fix image labels

javiereguiluz committed:
Merge remote-tracking branch 'upstream/6.0' into 6.0

* upstream/6.0:

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  remove display names from envelope headers

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  remove display names from envelope headers

dbu committed:
document default context for the serializer

javiereguiluz committed:
Merge branch '4.4' into 5.4

* 4.4:
  Improve the diff highlighting for custom webpack plugins

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  Improve the diff highlighting for custom webpack plugins

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  Improve the diff highlighting for custom webpack plugins

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  Improve the diff highlighting for custom webpack plugins

javiereguiluz committed:
Merge branch '4.4' into 5.4

* 4.4:
  [DependencyInjection] Add a section for testing service subscribers

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  [DependencyInjection] Add a section for testing service subscribers

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  [DependencyInjection] Add a section for testing service subscribers

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  [DependencyInjection] Add a section for testing service subscribers

javiereguiluz committed:
Merge branch '4.4' into 5.4

* 4.4:
  Promoting encore restart note

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  Promoting encore restart note

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  Promoting encore restart note

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  Promoting encore restart note

ThomasLandauer committed:
Removing self-closing slash from `<input>`s

I suggested the same for Symfony's templates, see https://github.com/symfony/symfony/pull/47715

dbu committed:
document serializer default_context

OskarStark committed:
feature #17310 document serializer default_context (dbu)

This PR was merged into the 5.4 branch.

Discussion
----------

document serializer default_context

fix https://github.com/symfony/symfony/issues/47183

add missing documentation for this serializer configuration option

Commits
-------

664bc4b79 document serializer default_context

OskarStark committed:
Merge branch '5.4' into 6.0

* 5.4:
  document serializer default_context

OskarStark committed:
Merge branch '6.0' into 6.1

* 6.0:
  document serializer default_context

OskarStark committed:
Merge branch '6.1' into 6.2

* 6.1:
  document serializer default_context

OskarStark committed:
Cleanup old versionadded directives

OskarStark committed:
minor #17312 Cleanup old versionadded directives (OskarStark)

This PR was merged into the 6.1 branch.

Discussion
----------

Cleanup old versionadded directives

Commits
-------

875662dd3 Cleanup old versionadded directives

OskarStark committed:
Merge branch '6.1' into 6.2

* 6.1:
  Cleanup old versionadded directives

wouterj committed:
feature #17284 [Security] Add documentation for programmatic login (johnkrovitch)

This PR was squashed before being merged into the 6.2 branch.

Discussion
----------

[Security] Add documentation for programmatic login

Add documentation for programmatic login

Fixes #16940

Commits
-------

b63ccf763 [Security] Add documentation for programmatic login

OskarStark committed:
feature #17253 Document the `framework.router.cache_dir` setting (mpdude)

This PR was squashed before being merged into the 6.2 branch.

Discussion
----------

Document the `framework.router.cache_dir` setting

This documents a new configuration setting suggested in https://github.com/symfony/symfony/pull/47496.

Commits
-------

24b9b1d0e Document the `framework.router.cache_dir` setting

mdoutreluingne committed:
[Messenger] Add new `messenger:stats` command that return a list of transports with their "to be processed" message count.

ayacoo committed:
[Console] Update console.rst

OskarStark committed:
feature #17295 [DependencyInjection] Document `stack` definitions (HeahDude)

This PR was merged into the 5.4 branch.

Discussion
----------

[DependencyInjection] Document `stack` definitions

Fixes #13587.

Commits
-------

02ae0a5da [DependencyInjection] Document `stack` definitions

OskarStark committed:
Merge branch '5.4' into 6.0

* 5.4:
  [DependencyInjection] Document `stack` definitions

OskarStark committed:
Merge branch '6.0' into 6.1

* 6.0:
  [DependencyInjection] Document `stack` definitions

OskarStark committed:
Merge branch '6.1' into 6.2

* 6.1:
  [DependencyInjection] Document `stack` definitions

JohJohan committed:
[Uuid] 17258 Add documentation for Uuid 7 and 8

javiereguiluz committed:
minor #17323 [Console] Update console.rst (ayacoo)

This PR was squashed before being merged into the 6.2 branch.

Discussion
----------

[Console] Update console.rst

Related #17307

Commits
-------

a0ee1ece4 [Console] Update console.rst

javiereguiluz committed:
Minor tweaks

javiereguiluz committed:
Merge branch '4.4' into 5.4

* 4.4:
  Mailer: remove port 99 for requestbin.com

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  Mailer: remove port 99 for requestbin.com
  [DependencyInjection] Document `stack` definitions

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  Mailer: remove port 99 for requestbin.com
  [DependencyInjection] Document `stack` definitions

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  Mailer: remove port 99 for requestbin.com
  [DependencyInjection] Document `stack` definitions

javiereguiluz committed:
minor #17322 [Messenger] Add new `messenger:stats` command that return a list of transports with their "to be processed" message count. (mdoutreluingne)

This PR was squashed before being merged into the 6.2 branch.

Discussion
----------

[Messenger] Add new `messenger:stats` command that return a list of transports with their "to be processed" message count.

Fixes #17085

Commits
-------

6d84eefcc [Messenger] Add new `messenger:stats` command that return a list of transports with their "to be processed" message count.

javiereguiluz committed:
Minor rewords

javiereguiluz committed:
minor #17315 [Uid] Add documentation for Uuid 7 and 8 (JohJohan)

This PR was merged into the 6.2 branch.

Discussion
----------

[Uid] Add documentation for Uuid 7 and 8

Adds documentation for Uuid  7 and 8

Fixes https://github.com/symfony/symfony-docs/issues/17258

Commits
-------

5222e73be [Uuid] 17258 Add documentation for Uuid 7 and 8

javiereguiluz committed:
Minor tweaks

javiereguiluz committed:
Minor typo

javiereguiluz committed:
Merge branch '4.4' into 5.4

* 4.4:
  [Kernel] Mention extra interfaces in `MicroKernel` section

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  [Kernel] Mention extra interfaces in `MicroKernel` section

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  [Kernel] Mention extra interfaces in `MicroKernel` section

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  [Kernel] Mention extra interfaces in `MicroKernel` section

javiereguiluz committed:
Merge branch '4.4' into 5.4

* 4.4:
  [Testing] Add a section to mock service dependencies

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  [Testing] Add a section to mock service dependencies

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  [Testing] Add a section to mock service dependencies

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  [Testing] Add a section to mock service dependencies

javiereguiluz committed:
minor #17305 improve documentation of default_context for the serializer (dbu)

This PR was merged into the 6.2 branch.

Discussion
----------

improve documentation of default_context for the serializer

additional info for https://github.com/symfony/symfony-docs/pull/17310

fix #17306

the code example will only work once https://github.com/symfony/symfony/pull/47243 is merged

Commits
-------

852aa117f document default context for the serializer

javiereguiluz committed:
Add the versionadded directive

OskarStark committed:
Merge remote-tracking branch 'upstream/6.0' into 6.0

* upstream/6.0:
  Minor tweak
  [Kernel] Mention extra interfaces in `MicroKernel` section
  Mailer: remove port 99 for requestbin.com
  [Testing] Add a section to mock service dependencies

OskarStark committed:
Merge remote-tracking branch 'upstream/6.1' into 6.1

* upstream/6.1:
  Minor tweak
  [Kernel] Mention extra interfaces in `MicroKernel` section
  Mailer: remove port 99 for requestbin.com
  [Testing] Add a section to mock service dependencies

OskarStark committed:
Merge remote-tracking branch 'upstream/6.2' into 6.2

* upstream/6.2:
  Add the versionadded directive
  Minor tweak
  [Kernel] Mention extra interfaces in `MicroKernel` section
  Minor typo
  Minor tweaks
  Minor rewords
  [Messenger] Add new `messenger:stats` command that return a list of transports with their "to be processed" message count.
  Minor tweaks
  [Console] Update console.rst
  [Uuid] 17258 Add documentation for Uuid 7 and 8
  Mailer: remove port 99 for requestbin.com
  [Testing] Add a section to mock service dependencies
  document default context for the serializer

OskarStark committed:
Remove unused versionadded directives

OskarStark committed:
Merge branch '6.0' into 6.1

* 6.0:
  Remove unused versionadded directives

OskarStark committed:
Merge branch '6.1' into 6.2

* 6.1:
  Remove unused versionadded directives

cafferata committed:
Fix typo

javiereguiluz committed:
minor #17325 Fix typo (cafferata)

This PR was merged into the 6.2 branch.

Discussion
----------

Fix typo

Fixed small typo (`sufix` = `suffix`).

Commits
-------

ab52db06f Fix typo

javiereguiluz committed:
minor #17325 Fix typo (cafferata)

This PR was merged into the 6.2 branch.

Discussion
----------

Fix typo

Fixed small typo (`sufix` = `suffix`).

Commits
-------

ab52db06f Fix typo

javiereguiluz committed:
Merge branch '4.4' into 5.4

* 4.4:
  Update serializer.rst

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  Update serializer.rst

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  Update serializer.rst

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  Update serializer.rst

javiereguiluz committed:
minor #17092 [Form] Better explaining what `getParent` actually does (ThomasLandauer)

This PR was merged into the 5.4 branch.

Discussion
----------

[Form] Better explaining what `getParent` actually does

Main reason for changing this is to stop presenting `getParent()` of being the "default" (or normal) way of doing it.

* I removed the "PHP extension" note-box, cause the (new) text for `getParent()` makes it clear what this does.
* I moved `finishView()` one paragraph up (didn't change anything)
* Moved all `getParent`-related stuff into its paragraph, and moved it to the top (since it's the most important question in the beginning)

Commits
-------

1c9b524c9 Better explaining what `getParent` actually does

javiereguiluz committed:
Minor tweak

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  Better explaining what `getParent` actually does

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  Better explaining what `getParent` actually does

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  Better explaining what `getParent` actually does

jeremyhalin committed:
docs(console): add caution for testing option

Add a caution about `InputOption::VALUE_NONE` testing.
Passing only a value without a key doesn't work and can lead to misunderstanding.

javiereguiluz committed:
minor #17288 [PropertyAccess] Document nullsafe operator usage (fsoedjede)

This PR was merged into the 6.2 branch.

Discussion
----------

[PropertyAccess] Document nullsafe operator usage

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/releases for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `6.x` for features of unreleased versions).

-->

Doc PR for https://github.com/symfony/symfony/pull/47511

Commits
-------

6a5d2eec0 [PropertyAccess] Document nullsafe operator usage

javiereguiluz committed:
Reword

javiereguiluz committed:
minor #17326 [Console] docs(console): add caution for testing option (jeremyhalin)

This PR was submitted for the 6.1 branch but it was merged into the 5.4 branch instead.

Discussion
----------

[Console] docs(console): add caution for testing option

Add a caution about `InputOption::VALUE_NONE` testing. Passing only a value without a key doesn't work and can lead to misunderstanding.

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/releases for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `6.x` for features of unreleased versions).

-->

Commits
-------

26a7fab00 docs(console): add caution for testing option

javiereguiluz committed:
Minor tweak

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  docs(console): add caution for testing option

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  docs(console): add caution for testing option

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  docs(console): add caution for testing option

javiereguiluz committed:
Merge branch '4.4' into 5.4

* 4.4:
  Adding all rpm commands

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  Adding all rpm commands

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  Adding all rpm commands

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  Adding all rpm commands

javiereguiluz committed:
minor #16999 [HttpKernel] Document the `terminate_on_cache_hit` option (javiereguiluz)

This PR was squashed before being merged into the 6.2 branch.

Discussion
----------

[HttpKernel] Document the `terminate_on_cache_hit` option

Fixes #16919.

Commits
-------

98a717a63 [HttpKernel] Document the `terminate_on_cache_hit` option

javiereguiluz committed:
minor #17327 [Serializer] Fix XML example of ignoring an attribute redux (likeuntomurphy)

This PR was merged into the 5.4 branch.

Discussion
----------

[Serializer] Fix XML example of ignoring an attribute redux

This effectively reopens #16824, which could not be reopened because of a rebase after the PR was closed.

Commits
-------

3c0734a7a Change 'ignore' element to attribute in Serializer XML example

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  minor #17325 Fix typo (cafferata)
  Change 'ignore' element to attribute in Serializer XML example

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  minor #17325 Fix typo (cafferata)
  Change 'ignore' element to attribute in Serializer XML example

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  minor #17325 Fix typo (cafferata)
  Change 'ignore' element to attribute in Serializer XML example

javiereguiluz committed:
minor #17308 [Security] Removing self-closing slash from `<input>`s (ThomasLandauer)

This PR was merged into the 6.2 branch.

Discussion
----------

[Security] Removing self-closing slash from `<input>`s

I suggested the same for Symfony's templates, see https://github.com/symfony/symfony/pull/47715

Commits
-------

c08a0a857 Removing self-closing slash from `<input>`s

javiereguiluz committed:
minor #17201 [DependencyInjection] Add information about debugging tagged services (BOB41K1987)

This PR was squashed before being merged into the 6.2 branch.

Discussion
----------

[DependencyInjection] Add information about debugging tagged services

Add documentations about debugging tagged services using `debug:container --tag`

Commits
-------

63c2a8d5b [DependencyInjection] Add information about debugging tagged services

javiereguiluz committed:
Minor reword

javiereguiluz committed:
Merge branch '4.4' into 5.4

* 4.4:
  Expanding on HTTP compression

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  Expanding on HTTP compression

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  Expanding on HTTP compression

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  Expanding on HTTP compression

javiereguiluz committed:
minor #17204 Usage of "Constructor Promotion" for PHP >= 8.0.2 (githubfromgui)

This PR was merged into the 6.0 branch.

Discussion
----------

Usage of "Constructor Promotion" for PHP >= 8.0.2

Symfony version [6.0 requires PHP 8.0.2](https://symfony.com/doc/6.0/setup.html#technical-requirements) or higher, so I would suggest to make usage of the [ `Constructor Promotion`](https://www.php.net/manual/en/language.oop5.decon.php#language.oop5.decon.constructor.promotion)  feature.

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/releases for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `6.x` for features of unreleased versions).

-->

Commits
-------

394815cb0 Usage of "Constructor Promotion" for PHP >= 8.0.2

javiereguiluz committed:
Fix code syntax

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  Usage of "Constructor Promotion" for PHP >= 8.0.2

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  Usage of "Constructor Promotion" for PHP >= 8.0.2

javiereguiluz committed:
minor #17067 Expanding recommendation for `finishView()`... (ThomasLandauer)

This PR was merged into the 5.4 branch.

Discussion
----------

Expanding recommendation for `finishView()`...

... to include the widgets created by ChoiceType as well.

Might conflict with a minor change I just did in https://github.com/symfony/symfony-docs/pull/17066

Commits
-------

48f0d4fe0 Expanding recommendation for `finishView()`...

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  Expanding recommendation for `finishView()`...

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  Expanding recommendation for `finishView()`...

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  Expanding recommendation for `finishView()`...

javiereguiluz committed:
minor #16720 [Messenger] Add doc for default routing for messages (clemherreman)

This PR was squashed before being merged into the 5.4 branch.

Discussion
----------

[Messenger] Add doc for default routing for messages

Hello,

I recently got bit on production after deploying a code change. We created a new `FooMessage` & `FooHandler`, but didn't update the `messenger.yaml` to include routing information.

This caused `FooMessage` to be directly handled by the HTTP process that dispatched the message, instead of the CLI, consumer process of Messenger. This lead to various issues: missing env var in HTTP context, lack of monitoring/stats as we only looks for messages metrics in the `consumer` container, not the `api` one :)

Looking at the code, I discovered you may define a default transport, but couldn't find it in the doc.

Commits
-------

955b39345 [Messenger] Add doc for default routing for messages

javiereguiluz committed:
Tweaks

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  [Messenger] Add doc for default routing for messages

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  [Messenger] Add doc for default routing for messages

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  [Messenger] Add doc for default routing for messages

qdequippe committed:
Update testing.rst

Co-authored-by: Jules Pietri <heahdude@yahoo.fr>

qdequippe committed:
Update testing.rst

Co-authored-by: Jules Pietri <heahdude@yahoo.fr>

qdequippe committed:
Update testing.rst

Co-authored-by: Jules Pietri <heahdude@yahoo.fr>

andre-pdassis committed:
Add missing space

OskarStark committed:
minor #17330 Add missing space (decocodes)

This PR was submitted for the 6.2 branch but it was merged into the 6.1 branch instead.

Discussion
----------

Add missing space

Commits
-------

b636d2e2d Add missing space

OskarStark committed:
Merge branch '6.1' into 6.2

* 6.1:
  Add missing space

qdequippe committed:
add break lines and remove useless line

OskarStark committed:
minor #16750 Add caution message about reset container on each request (Quentin Dequippe, qdequippe)

This PR was merged into the 5.4 branch.

Discussion
----------

Add caution message about reset container on each request

Thanks to SymfonyCasts explication https://symfonycasts.com/screencast/api-platform-security/test-roles-reload :)

Commits
-------

1898849ad add break lines and remove useless line
dad267e2c Update testing.rst
c4c189979 Update testing.rst
6f431a4bf Update testing.rst
4dcabf935 Add caution message about reset container on each request

OskarStark committed:
Merge branch '5.4' into 6.0

* 5.4:
  add break lines and remove useless line
  Update testing.rst
  Update testing.rst
  Update testing.rst
  Add caution message about reset container on each request

OskarStark committed:
Merge branch '6.0' into 6.1

* 6.0:
  add break lines and remove useless line
  Update testing.rst
  Update testing.rst
  Update testing.rst
  Add caution message about reset container on each request

OskarStark committed:
Merge branch '6.1' into 6.2

* 6.1:
  add break lines and remove useless line
  Update testing.rst
  Update testing.rst
  Update testing.rst
  Add caution message about reset container on each request

wouterj committed:
minor #16799 [Security] Correcting bugs for provider and JSON login (KosticDusan4D)

This PR was merged into the 6.0 branch.

Discussion
----------

[Security] Correcting bugs for provider and JSON login

On lines 501 and 502 I propose this change, because it confused me so much which provider to use, and in the case that you are explaining there you should use earlier configured app_user_provider, not users_in_memory.

On line 1043 I had a problem that $user is always null until I deleted the "?User" from parameters. I think that PHP gets confused here with the way it is written. And to say that now with that part removed everything works fine.

Thanks in advance and I hope that I contributed even a little as this is.

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/releases for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `6.x` for features of unreleased versions).

-->

Commits
-------

2309b0b81 Correcting bugs for provider and JSON login

javiereguiluz committed:
Merge branch '4.4' into 5.4

* 4.4:
  Use fourth argument of RoutingConfigurator instead of non-existent method

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  Use fourth argument of RoutingConfigurator instead of non-existent method

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  Use fourth argument of RoutingConfigurator instead of non-existent method
  Correcting bugs for provider and JSON login

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  Use fourth argument of RoutingConfigurator instead of non-existent method
  Correcting bugs for provider and JSON login

javiereguiluz commented:
Let's ask @HeahDude, our Symfony Forms expert, to see if he can review this proposal. Thanks!

mdoutreluingne committed:
[Messenger] Deprecate `MessageSubscriberInterface`

mdoutreluingne committed:
[Messenger] Removing note docblock for MessageHandlerInterface

The docblock note is no longer required from Symfony 6.0 onwards as it requires PHP 8.0.

AboulhajHakim committed:
[IntegerType] fix rounding_mode default value

change default value of rounding_mode option from \NumberFormatter::ROUND_HALFUP to default: \NumberFormatter::ROUND_DOWN 
see https://github.com/symfony/symfony/blob/6.0/src/Symfony/Component/Form/Extension/Core/Type/IntegerType.php::49

wouterj committed:
Adding full subscriber example

Reason: It's not so easy to figure this out, since the linked Event Subscriber page doesn't show how to subscribe to this specific event.

Questions:
* The introduction text says "e.g. invalidate some tokens". How can this be done?
* How can you add a flash message? `$this->addFlashMessage()` didn't work for me.
* I'm extending `AbstractController` to have access to `$this->generateUrl()` - is this the easiest/best way?

wouterj committed:
minor #15186 Adding full subscriber example (ThomasLandauer)

This PR was submitted for the 5.2 branch but it was merged into the 5.4 branch instead.

Discussion
----------

Adding full subscriber example

Reason: It's not so easy to figure this out, since the linked Event Subscriber page doesn't show how to subscribe to this specific event.

Questions:
* The introduction text says "e.g. invalidate some tokens". How can this be done?
* How can you add a flash message? `$this->addFlash()` didn't work for me.
* I'm extending `AbstractController` to have access to `$this->generateUrl()` - is this the easiest/best way?

Commits
-------

4f90b3cba Adding full subscriber example

wouterj committed:
[#15186] Update example

wouterj committed:
[#15428] Add a bit more detail to bootstrap article

wouterj committed:
minor #15428 Fix example code of customization of bootstrapping in test (sudo-barun)

This PR was submitted for the 5.3 branch but it was merged into the 5.4 branch instead.

Discussion
----------

Fix example code of customization of bootstrapping in test

The change is copied from Symfony 4.4.

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/releases for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `5.x` for features of unreleased versions).

-->

Commits
-------

a7c73785b [#15428] Add a bit more detail to bootstrap article
c08e8fd15 Fix example code of customization of bootstrapping in test

wouterj committed:
Merge branch '4.4' into 5.4

* 4.4:
  Example remote ElasticsearchLogstashHandler
  [Validator] Add hint for testing custom constraints

wouterj committed:
Document new Logstash handler argument

wouterj committed:
Merge branch '5.4' into 6.0

* 5.4:
  Document new Logstash handler argument
  Example remote ElasticsearchLogstashHandler
  [#15428] Add a bit more detail to bootstrap article
  Fix example code of customization of bootstrapping in test
  [#15186] Update example
  Adding full subscriber example
  [Validator] Add hint for testing custom constraints

wouterj committed:
Remove 5.4 versionadded

wouterj committed:
Merge branch '6.0' into 6.1

* 6.0:
  Remove 5.4 versionadded
  Document new Logstash handler argument
  Example remote ElasticsearchLogstashHandler
  [#15428] Add a bit more detail to bootstrap article
  Fix example code of customization of bootstrapping in test
  [#15186] Update example
  Adding full subscriber example
  [Validator] Add hint for testing custom constraints

wouterj committed:
Merge branch '6.1' into 6.2

* 6.1:
  Remove 5.4 versionadded
  Document new Logstash handler argument
  Example remote ElasticsearchLogstashHandler
  [#15428] Add a bit more detail to bootstrap article
  Fix example code of customization of bootstrapping in test
  [#15186] Update example
  Adding full subscriber example
  [Validator] Add hint for testing custom constraints

javiereguiluz committed:
Merge branch '4.4' into 5.4

* 4.4:
  Minor (language) improvements

javiereguiluz committed:
minor #17333 [IntegerType] fix rounding_mode default value (hakimtechmyteam)

This PR was submitted for the 6.0 branch but it was merged into the 5.4 branch instead.

Discussion
----------

[IntegerType] fix rounding_mode default value

change default value of rounding_mode option from \NumberFormatter::ROUND_HALFUP to default: \NumberFormatter::ROUND_DOWN  see https://github.com/symfony/symfony/blob/6.0/src/Symfony/Component/Form/Extension/Core/Type/IntegerType.php

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/releases for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `6.x` for features of unreleased versions).

-->

Commits
-------

4238c6d85 [IntegerType] fix rounding_mode default value

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  [IntegerType] fix rounding_mode default value
  Minor (language) improvements

javiereguiluz committed:
minor #17332 [Messenger] Removing note docblock for `MessageHandlerInterface` (mdoutreluingne)

This PR was merged into the 6.0 branch.

Discussion
----------

[Messenger] Removing note docblock for `MessageHandlerInterface`

The docblock note is no longer necessary  required from Symfony 6.0 onwards as it requires PHP 8.0.

What do you think?

Commits
-------

74daa0e56 [Messenger] Removing note docblock for MessageHandlerInterface

javiereguiluz committed:
minor #17331 [Messenger] Deprecate `MessageSubscriberInterface` (mdoutreluingne)

This PR was squashed before being merged into the 6.2 branch.

Discussion
----------

[Messenger] Deprecate `MessageSubscriberInterface`

Fixes #17319

For `MessageHandlerInterface` class, it's not necessary to create a deprecation directive because we display it more since Symfony 5.3 in Symfony doc. I have created a PR #17332 for removing note block as it is no longer necessary in Symfony 6.0.

Commits
-------

cb957d76a [Messenger] Deprecate `MessageSubscriberInterface`

javiereguiluz committed:
Merge branch '4.4' into 5.4

* 4.4:
  docs: Add the mime type in the example

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  docs: Add the mime type in the example

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  docs: Add the mime type in the example
  [IntegerType] fix rounding_mode default value
  Minor (language) improvements
  [Messenger] Removing note docblock for MessageHandlerInterface

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  docs: Add the mime type in the example
  [IntegerType] fix rounding_mode default value
  Minor (language) improvements
  [Messenger] Removing note docblock for MessageHandlerInterface

ogizanagi committed:
[Serializer] Allow to provide (de)normalization context in mapping

alamirault committed:
[Serializer] Add PropertyNormalizer::NORMALIZE_VISIBILITY documentation

javiereguiluz committed:
Merge branch '4.4' into 5.4

* 4.4:
  [Symfony UX] reference React component in React page
  [Symfony UX] reference Vue component in VueJs page
  Reword "max lifetime" to "default lifetime"

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  [Symfony UX] reference React component in React page
  [Symfony UX] reference Vue component in VueJs page
  Reword "max lifetime" to "default lifetime"

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  [Symfony UX] reference React component in React page
  [Symfony UX] reference Vue component in VueJs page
  Reword "max lifetime" to "default lifetime"

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  [Symfony UX] reference React component in React page
  [Symfony UX] reference Vue component in VueJs page
  Reword "max lifetime" to "default lifetime"

HeahDude commented on form/form_themes.rst:
```diff
@@ -371,6 +371,65 @@ by ``row``, ``widget``, ``label``, or ``help``):
         {# the field of TaskType #}
     {% endblock %}
 
+Access data in a Form Choice
```
This section here is breaking the flow of this form themes page IMHO.
Maybe we should have it at the end of the [ChoiceType reference page](https://symfony.com/doc/current/reference/forms/types/choice.html#field-variables) with a link from here?

HeahDude commented on form/form_themes.rst:
```diff
@@ -371,6 +371,65 @@ by ``row``, ``widget``, ``label``, or ``help``):
+Access data in a Form Choice
+.............................
+
+When you have a ChoiceType or an EntityType, and you want to customize the ``entry``, row or widget, you can also retrieve
```
```suggestion
When you use an expanded ``ChoiceType`` and need to customize the children ``entry`` blocks, 
```
Then `EntityType` and other children (i.e. `LocaleType`) could have a link to this section and we could also link back to the `entry` section of form themes.

xabbuh committed:
drop the MessageSubscriberInterface in favor of the #AsMessageHandler attribute

HeahDude commented on form/form_themes.rst:
```diff
@@ -371,6 +371,65 @@ by ``row``, ``widget``, ``label``, or ``help``):
+.............................
+
+When you have a ChoiceType or an EntityType, and you want to customize the ``entry``, row or widget, you can also retrieve
+the data of the choice, but not directly in the ``form.vars.data`` of the child. See an exemple::
```
We could improve this sentence altogether by defining what is actually the entry data, before suggesting to use `form.parent.vars.choices`, here's a possible alternative:
```rst
When you use an expanded ``ChoiceType`` and need to customize the children ``entry`` blocks,
the ``form.vars`` of entries (radio button or checkbox) may not be enough since each holds a
boolean value meaning whether a choice is selected or not.
To get the full list of choices data and values, you will need to access the ``choices`` variable
from their parent form (the ``ChoiceType`` itself) with ``form.parent.vars.choices``:
```

HeahDude commented on form/form_themes.rst:
```diff
@@ -371,6 +371,65 @@ by ``row``, ``widget``, ``label``, or ``help``):
+When you have a ChoiceType or an EntityType, and you want to customize the ``entry``, row or widget, you can also retrieve
+the data of the choice, but not directly in the ``form.vars.data`` of the child. See an exemple::
+
+    class TaskManagerType extends AbstractType
```
By moving this section as suggested above, adding one more PHP example should not be necessary in that context, a reference to https://symfony.com/doc/4.4/reference/forms/types/choice.html#advanced-example-with-objects could be enough as it is also used to illustrate most of the options in that page.

HeahDude commented on form/form_themes.rst:
```diff
@@ -371,6 +371,65 @@ by ``row``, ``widget``, ``label``, or ``help``):
+        }
+    }
+
+Now we want to display each checkbox inside a table element. The entity is not accessible in ``form.vars.value`` nor ``form.vars.data``.
```
Another alternative for the following paragraphs could be:
```rst
Given the advanced object example, each entry would have access to the following variables:

.. code-block:: html+twig

    {# `true` or `false`, whether the current choice is selected as radio or checkbox #}
    {{ form.vars.data }}

    {# the current choice value (i.e a category name when `'choice_value' => 'name'` #}
    {{ form.vars.value }}

    {# a map of `ChoiceView` or `ChoiceGroupView` instances indexed by choice values or group names #}
    {{ form.parent.vars.choices }}
```

HeahDude reviewed (commented):
Thanks for that PR!
I agree we should better document the theming of choice entries but I'm not convince it should be done in the current way.
I've left some ideas, what do you think @spike31 @javiereguiluz?

eerison committed:
Update choice_translation_parameters.rst.inc

Adding missing ";" for the `choice_translation_parameters` example

javiereguiluz committed:
Merge branch '4.4' into 5.4

* 4.4:
  [Form] Fix `selectedchoice` test link
  [Form] Improve `ChoiceType` CS

javiereguiluz committed:
minor #17354 Update choice_translation_parameters.rst.inc (eerison)

This PR was submitted for the 6.2 branch but it was merged into the 5.4 branch instead.

Discussion
----------

Update choice_translation_parameters.rst.inc

Adding missing ";" for the `choice_translation_parameters` example

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/releases for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `6.x` for features of unreleased versions).

-->

Commits
-------

c84504878 Update choice_translation_parameters.rst.inc

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  Update choice_translation_parameters.rst.inc
  [Form] Fix `selectedchoice` test link
  [Form] Improve `ChoiceType` CS

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  Update choice_translation_parameters.rst.inc
  [Form] Fix `selectedchoice` test link
  [Form] Improve `ChoiceType` CS

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  Update choice_translation_parameters.rst.inc
  [Form] Fix `selectedchoice` test link
  [Form] Improve `ChoiceType` CS

javiereguiluz committed:
minor #17350 [Messenger] drop the MessageSubscriberInterface in favor of the #AsMessageHandler attribute (xabbuh)

This PR was merged into the 6.2 branch.

Discussion
----------

[Messenger] drop the MessageSubscriberInterface in favor of the #AsMessageHandler attribute

Commits
-------

157020f92 drop the MessageSubscriberInterface in favor of the #AsMessageHandler attribute

javiereguiluz committed:
Minor tweak

wouterj committed:
[#17328] Minor changes

wouterj committed:
feature #17328  docs: add docs for programmatic logout (johnkrovitch)

This PR was merged into the 6.2 branch.

Discussion
----------

 docs: add docs for programmatic logout

This PR adds documentation for the new feature "programmatic logout" merge in the PR https://github.com/symfony/symfony/pull/41406

Commits
-------

ac46df4dd [#17328] Minor changes
5611f8821 docs: add docs for programmatic logout

javiereguiluz committed:
Merge branch '4.4' into 5.4

* 4.4:
  Readded the link to Akamai
  Remove akamai broken link

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  Readded the link to Akamai
  Remove akamai broken link

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  Readded the link to Akamai
  Remove akamai broken link

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  Readded the link to Akamai
  Remove akamai broken link

javiereguiluz committed:
minor #17347 [Serializer] Add PropertyNormalizer::NORMALIZE_VISIBILITY documentation (alamirault)

This PR was merged into the 6.2 branch.

Discussion
----------

[Serializer] Add PropertyNormalizer::NORMALIZE_VISIBILITY documentation

Try documenting #17036

Commits
-------

85ad399b9 [Serializer] Add PropertyNormalizer::NORMALIZE_VISIBILITY documentation

javiereguiluz committed:
Reword

javiereguiluz committed:
minor #17270 Fix the code example leading to an error (qzminski)

This PR was squashed before being merged into the 6.1 branch.

Discussion
----------

Fix the code example leading to an error

Commits
-------

8bf5f9e81 Fix the code example leading to an error

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  Fix the code example leading to an error

javiereguiluz committed:
Merge branch '4.4' into 5.4

* 4.4:
  docs(test): add array example for options

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  docs(test): add array example for options

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  docs(test): add array example for options

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  docs(test): add array example for options

javiereguiluz committed:
Merge branch '4.4' into 5.4

* 4.4:
  Mentioning svelte as aa valid loader override type
  Force SimpleBus https url
  Fix Stumptown Syndicate guideline broken link

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  Mentioning svelte as aa valid loader override type
  Force SimpleBus https url
  Fix Stumptown Syndicate guideline broken link

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  Mentioning svelte as aa valid loader override type
  Force SimpleBus https url
  Fix Stumptown Syndicate guideline broken link

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  Mentioning svelte as aa valid loader override type
  Force SimpleBus https url
  Fix Stumptown Syndicate guideline broken link

javiereguiluz committed:
Merge branch '4.4' into 5.4

* 4.4:
  Change secure.php.net link to php.net
  [Validator] Combine #13898 with recent changes
  [Serializer] Remove note about installing the FrameworkExtraBundle for using annotations
  [Validator] Unit Tests

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  Change secure.php.net link to php.net
  [Validator] Combine #13898 with recent changes
  [Serializer] Remove note about installing the FrameworkExtraBundle for using annotations
  [Validator] Unit Tests

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  Change secure.php.net link to php.net
  [Validator] Combine #13898 with recent changes
  [Serializer] Remove note about installing the FrameworkExtraBundle for using annotations
  [Validator] Unit Tests

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  Change secure.php.net link to php.net
  [Validator] Combine #13898 with recent changes
  [Serializer] Remove note about installing the FrameworkExtraBundle for using annotations
  [Validator] Unit Tests

javiereguiluz committed:
minor #17136 [RateLimiter] Adding info about clearing cache (ThomasLandauer)

This PR was merged into the 5.4 branch.

Discussion
----------

[RateLimiter] Adding info about clearing cache

I'm just *guessing* this - please double-check!

Commits
-------

178a6cd6d Adding info about clearing cache

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  Adding info about clearing cache

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  Adding info about clearing cache

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  Adding info about clearing cache

wouterj committed:
[HttpKernel] Replace ArgumentValueResolverInterface by ValueResolverInterface

nlemoine committed:
Fix `notifier.flash_message_importance_mapper` service definition

welcoMattic committed:
Add new cases covered by PhpAstExtractor

ker0x committed:
[Notifier] Update FreeMobile DSN

hiddewie committed:
Use `addPart` instead of `embed*` or `attach*`.

javiereguiluz committed:
Merge branch '4.4' into 5.4

* 4.4:
  Correct time type return
  Fix a typo
  Reword
  add tip about listener priority to mailer.rst

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  Correct time type return
  Fix a typo
  Reword
  add tip about listener priority to mailer.rst

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  Correct time type return
  Fix a typo
  Reword
  add tip about listener priority to mailer.rst

javiereguiluz committed:
minor #17363 [Notifier] Fix `notifier.flash_message_importance_mapper` service definition (nlemoine)

This PR was merged into the 6.1 branch.

Discussion
----------

[Notifier] Fix `notifier.flash_message_importance_mapper` service definition

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/releases for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `6.x` for features of unreleased versions).

-->

Commits
-------

5dbace5e7 Fix `notifier.flash_message_importance_mapper` service definition

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  Correct time type return
  Fix `notifier.flash_message_importance_mapper` service definition
  Fix a typo
  Reword
  add tip about listener priority to mailer.rst

javiereguiluz committed:
minor #17371 [Notifier] Update FreeMobile DSN (ker0x)

This PR was merged into the 5.4 branch.

Discussion
----------

[Notifier] Update FreeMobile DSN

PR to reflect changes from [symfony/symfony/47915](https://github.com/symfony/symfony/pull/47915)

Commits
-------

b532f3751 [Notifier] Update FreeMobile DSN

javiereguiluz committed:
minor #17345 [Serializer] Allow to provide (de)normalization context in mapping (ogizanagi)

This PR was merged into the 5.4 branch.

Discussion
----------

[Serializer] Allow to provide (de)normalization context in mapping

Fixes #14988

Commits
-------

800017d42 [Serializer] Allow to provide (de)normalization context in mapping

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  [Notifier] Update FreeMobile DSN
  [Serializer] Allow to provide (de)normalization context in mapping

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  [Notifier] Update FreeMobile DSN
  [Serializer] Allow to provide (de)normalization context in mapping

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  [Notifier] Update FreeMobile DSN
  [Serializer] Allow to provide (de)normalization context in mapping

javiereguiluz committed:
minor #17135 [Security] Adding info where login attempts are stored (ThomasLandauer)

This PR was squashed before being merged into the 5.4 branch.

Discussion
----------

[Security] Adding info where login attempts are stored

Commits
-------

7ad5d9f5b [Security] Adding info where login attempts are stored

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  [Security] Adding info where login attempts are stored

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  [Security] Adding info where login attempts are stored

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  [Security] Adding info where login attempts are stored

ogizanagi committed:
[Serializer] Fixup attributes import

javiereguiluz committed:
minor #17379 [Serializer] Fixup attributes import (ogizanagi)

This PR was merged into the 5.4 branch.

Discussion
----------

[Serializer] Fixup attributes import

Re-reading the rendered doc at https://symfony.com/doc/5.4/serializer.html#serializer-context, spotted I missed this.

Commits
-------

34881a8ab [Serializer] Fixup attributes import

javiereguiluz committed:
Merge branch '5.4' into 6.0

* 5.4:
  [Serializer] Fixup attributes import

javiereguiluz committed:
Merge branch '6.0' into 6.1

* 6.0:
  [Serializer] Fixup attributes import

javiereguiluz committed:
Merge branch '6.1' into 6.2

* 6.1:
  [Serializer] Fixup attributes import

javiereguiluz committed:
minor #17369 [Translation] Add new cases covered by PhpAstExtractor (welcoMattic)

This PR was merged into the 6.2 branch.

Discussion
----------

[Translation] Add new cases covered by PhpAstExtractor

In https://github.com/symfony/symfony/pull/46161, we rewrite the PHP Translation extractor. It covers more cases, like named arguments.
Documentation needed to be updated consequently.

Commits
-------

5029cb228 Add new cases covered by PhpAstExtractor

javiereguiluz committed:
Add the versionadded directive

javiereguiluz committed:
minor #17362 [HttpKernel] Replace ArgumentValueResolverInterface by ValueResolverInterface (wouterj)

This PR was merged into the 6.2 branch.

Discussion
----------

[HttpKernel] Replace ArgumentValueResolverInterface by ValueResolverInterface

Fixes #17225

I've also updated the example using a different example (one we use frequently in our projects). The issue with the current example is that it re-implements something that was already built-in, requiring all sorts of workarounds.

cc `@nicolas`-grekas

Commits
-------

b6bcc3d0e [HttpKernel] Replace ArgumentValueResolverInterface by ValueResolverInterface

javiereguiluz committed:
Tweaks

javiereguiluz committed:
minor #17372 Use `addPart` instead of `embed*` or `attach*`. (hiddewie)

This PR was squashed before being merged into the 6.2 branch.

Discussion
----------

Use `addPart` instead of `embed*` or `attach*`.

Fixes https://github.com/symfony/symfony-docs/issues/17303
Ref https://github.com/symfony/symfony-docs/issues/17353
Ref https://github.com/symfony/symfony/pull/47711
Ref https://github.com/symfony/symfony/pull/47462

This PR contains the documentation for simplifications for adding parts to `Email`s.

Commits
-------

ebbffc551 Use `addPart` instead of `embed*` or `attach*`.

javiereguiluz committed:
Minor tweak

gilles-g committed:
Merge branch 'symfony:6.2' into form_theme_customize_choice

gilles-g commented:
Hum, never ever click on "sync branch in your fork", I will recreate a PR with your recommendations @HeahDude  :D

HeahDude commented:
@spike31 if you have `symfony` as a remote you can try `git pull --rebase symfony 4.4` or `git rebase -i symfony/4.4`.
